### ISSUE TITLE
Error trying to deserialize with bytes_to_var_with_objects

### ISSUE THREAD

**Issue by blackears** [godotengine/godot#80229]
### Godot version

v4.1.stable.official [970459615]

### System information

Windows 10, Compatability

### Issue description

I'm trying to turn an object into a PackedByteArray so I can send it across an RPC call.  However, when I try to deserialize, it I get an error saying `ParserError: Class "Foo" hides a global script class`.  This only happens if I include the line with `bytes_to_var_with_objects` in the script.

### Steps to reproduce

Just run the following lines.  This seems to be causing the error:

```
	var foo:Foo = Foo.new()
	foo.alpha = 6
	foo.beta = 5
	
	var bytes:PackedByteArray = var_to_bytes_with_objects(foo)
	var foo_clone:Foo = bytes_to_var_with_objects(bytes)
```

This is Foo:
```
extends Resource
class_name Foo

var alpha:int
var beta:float
```

### Minimal reproduction project

[objectSerialization.zip](https://github.com/godotengine/godot/files/12255273/objectSerialization.zip)

**Comment by AThousandShips** [comment]
Looks like the same issue as:
* #68666

**Comment by AimiIsFat** [comment]
This is still an issue and can make some things more complex, especially networking, which uses bytes_to_var_with_objects in put_var if objects are allowed.

Here's a simple reproduction script if it helps:
```py
class_name Foo
extends Node

func _ready() -> void:
	var dict = {object = self}
	
	var encoded = var_to_bytes_with_objects(dict)
	
	var object = bytes_to_var_with_objects(encoded)
```
Hoping this gets fixed soon. Will probably just use var_to_str and str_to_var on all objects

**Comment by andzejsp** [comment]
no fix yet ? :(

**Comment by AThousandShips** [comment]
Please try [4.3.dev6](https://godotengine.org/article/dev-snapshot-godot-4-3-dev-6/), this might have been fixed assuming it's the same as #68666

**Comment by andzejsp** [comment]
> Please try [4.3.dev6](https://godotengine.org/article/dev-snapshot-godot-4-3-dev-6/), this might have been fixed assuming it's the same as #68666

Wow, nice, finally can use this in rpc() functions. 

Thanks, keep up the Lords work!!

**Comment by dalexeev** [comment]
* Duplicate of #68666 (uses the same mechanism under the hood).
* Fixed by #78219 and #90693 for future 4.3.

Thanks for reporting the bug nonetheless!
